src/events/channels/transactionsChannel.ts

```diff
@@ -0,0 +1,167 @@
+import EventEmitter from 'events'
+import { AnyTransaction, WebSocketClient } from 'adamant-api'
+import { schedule } from 'node-cron'
+import { logger } from '../../modules/logger.js'
+import { config } from '../../config/index.js'
+import { adamantClient } from '../../modules/adamantClient.js'
+import { prisma } from '../../modules/prisma.js'
+import { JobName } from '@prisma/client'
+import { isSignalTx } from '../../services/parser/isSignalTx.js'
+import { isTxToNotify } from '../../services/parser/isTxToNotify.js'
+import { ChatMessageTransaction } from 'adamant-api/dist/api/generated.js'
+
+class TransactionsChannel extends EventEmitter {
+  // Specify types of events
+  override emit(event: 'newSignalMessage', tx: ChatMessageTransaction): boolean
+  override emit(event: 'newMessage', tx: AnyTransaction): boolean
+  override emit(event: string, ...args: never[]): boolean {
+    return super.emit(event, ...args)
+  }
+
+  override on(
+    event: 'newSignalMessage',
+    listener: (tx: ChatMessageTransaction) => void
+  ): this
+  override on(event: 'newMessage', listener: (tx: AnyTransaction) => void): this
+  override on(event: string, listener: (...args: any[]) => void): this {
+    return super.on(event, listener)
+  }
+
+  private isLocked: boolean
+  private processedTxs: { [key: string]: AnyTransaction } = {} // cache for processed transactions
+  adamantSocket?: WebSocketClient
+
+  constructor() {
+    super()
+    this.isLocked = false
+  }
+
+  initSocket() {
+    adamantClient.initSocket({
+      wsType: 'ws',
+      admAddress: config.adamantAccount.address
+    })
+    logger.info(
+      `Adamant Client socket initialized on ${config.adamantAccount.address} address`
+    )
+
+    if (adamantClient.socket) {
+      adamantClient.socket.on((tx) => {
+        this.handleTransaction(tx)
+      })
+      adamantClient.socket.catch((error) => logger.error(error))
+
+      this.adamantSocket = adamantClient.socket
+    }
+  }
+
+  startJob() {
+    logger.info(
+      `Spawned transaction parser job with ${config.app.txCheckInterval} interval`
+    )
+    return schedule(config.app.txCheckInterval, async () => {
```

Comment:
Instead of using `schedule`, you can use a regular `setTimeout` that invokes itself after each iteration. In this case, you can remove the `this.isLocked` check, as invoking the callback multiple times won't be possible.

Reply:
Additionally add `destroy` method to this class that will:

1. Stop the timer
2. Stop listening for adamantClient events (`adamantClient.socket.off(callback)`)

Invoke `transactionsChannel.destroy()` from main.ts on SIGINT/SIGTERM